### PR TITLE
Bugfix - client_details styling of validation messages

### PR DESCRIPTION
**Pull request by willc-work**
Displays inline error messages and adds relevant formatting

[Link to story](https://dsdmoj.atlassian.net/browse/AP-11)

Client first and last name show the error message between the label and form and the field is 
highlighted as per gds requirements.

client dob is validated and relevant error messages are shown in the summary and inline.
The dob area is highlighted when an invalid or future date is entered

## Checklist

- [x] Tests and rubocop should be passing: `bundle exec rake`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely. 
- [x] The PR description should say what you changed and why, with a link to the JIRA story.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/views/client_details/_form.html.erb [hunk 1]
```diff
@@ -1,6 +1,3 @@
-<% content_for :navigation do %>
```
Comment (lostie) [inline app/views/client_details/_form.html.erb]: Is this meant to be removed on your PR? This was a new feature added very recently by @willmcb if I'm not mistaken.

File: app/models/client_detail.rb [hunk 2]
```diff
@@ -9,40 +9,23 @@ class ClientDetail
...
-      errors.add(:dob_year, "Year cannot be more than 115 years ago")
+  def parse_date
+    if !Date.valid_civil?(dob_year.to_i,dob_month.to_i,dob_day.to_i)
+      errors.add(:dob, "Enter a valid date of birth")
+    end
+  end
+
+  def validate_date_of_birth
+    if dob.present? && dob > Date.today.to_s
+      errors.add(:dob, 'Date of birth must be in the past')
```
Comment (lostie) [inline app/models/client_detail.rb]: **[not a blocker]** Consider moving the copy of this errors to the [locales file](https://github.com/ministryofjustice/laa-apply-for-legal-aid-provider-frontend/blob/master/config/locales/en.yml#L15) and just defining the error with a code:

```ruby
errors.add(:dob, :not_in_past)
```
and somewhere in the locale:

```ruby
...
        client_detail:
          attributes:
              first_name:
                blank: 'Enter first name'
              last_name:
                blank: 'Enter last name'
              dob:
                not_in_past: 'Date of birth must be in the past'
```

File: app/views/client_details/_form.html.erb [hunk 3]
```diff
@@ -43,61 +40,100 @@
...
-        <div class="govuk-form-group">
-          <%= form.label :dob_month, :class =>"govuk-label govuk-date-input__label" %>
-          <%= form.text_field :dob_month, id: 'dob_month', :class =>"govuk-input govuk-date-input__input govuk-input--width-2" %>
-        </div>
-      </div>
-      <div class="govuk-date-input__item">
-        <div class="govuk-form-group">
-          <%= form.label :dob_year, :class =>"govuk-label govuk-date-input__label" %>
-          <%= form.text_field :dob_year, id: 'dob_year', :class =>"govuk-input govuk-date-input__input govuk-input--width-4" %>
+        <% if @client_detail.errors[:first_name].any? %>
```
Comment (lostie) [inline app/views/client_details/_form.html.erb]: ⚠️ I think this needs to be revisited as we seem to be doing a lot of custom code to display errors, whereas Rails should be doing most of that heavy work for us as it has that already in place.

If you're trying to customise how, in general all the fields with errors should be displayed, then what you can do is something like this: http://trevorelwell.me/customize-field-errors-with-rails-5-bootstrap-4/

Which basically allows you to overwrite the method Rails uses to render the errors.

Happy to talk offline about this, but I would really like to see this view a lot DRYer than it is at the moment.

File: app/assets/stylesheets/application.scss [hunk 4]
```diff
@@ -23,9 +23,9 @@
 
  @import "govuk-frontend/all";
 
-.field_with_errors input{
-  border: 4px solid red !important;
-}
+ .field_with_errors input{
```
[the remark this answers is not in the document]
Comment (willc-work) [reply app/assets/stylesheets/application.scss]: using field_with_errors to highlight the input field means that i can leave the text input fields out of the if/else statements and still apply the relevant styles where required. I may be able to remove this when error display is moved to a helper class

**Pull request merged**